app/src/main/java/com/example/bigowlapp/utils/ScheduledLocationTrackingManager.java

```diff
@@ -0,0 +1,81 @@
+
+public class ScheduledLocationTrackingManager {
+
+    private static final int DEFAULT_TRACKING_RADIUS_METERS = 300;
+    private static final long DEFAULT_TRACKING_TIME_MILLISECONDS = 30L * Constants.MINUTE_TO_MILLISECONDS;
+    // use 0 for instant response, and allow delay for better battery life
+    private static final int DEFAULT_MAX_NOTIFY_DELAY_MILLISECONDS = 5 * Constants.MINUTE_TO_MILLISECONDS;
+
+    private final Context context;
+    private final GeofencingClient geofencingClient;
+    private final FusedLocationProviderClient locationClient;
+    private PendingIntent geofencePendingIntent;
+
+    public ScheduledLocationTrackingManager(Context context) {
+        this.context = context;
+        geofencingClient = LocationServices.getGeofencingClient(context);
+        locationClient = LocationServices.getFusedLocationProviderClient(context);
+    }
+
+    public Task<Void> addNewScheduledLocationToTrack(Schedule scheduleWithLocationToTrack) {
+        if (ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            return Tasks.forException(new SecurityException("Requires " + Manifest.permission.ACCESS_FINE_LOCATION + " permission."));
+        }
+
+        return locationClient.getCurrentLocation(LocationRequest.PRIORITY_HIGH_ACCURACY, null)
+                .continueWithTask(locTask -> geofencingClient.addGeofences(
+                        buildRequestToTrack(scheduleWithLocationToTrack), getGeofencePendingIntent()));
+    }
+
+    private GeofencingRequest buildRequestToTrack(Schedule scheduleWithLocationToTrack) {
+        return new GeofencingRequest.Builder()
+                .setInitialTrigger(GeofencingRequest.INITIAL_TRIGGER_ENTER | GeofencingRequest.INITIAL_TRIGGER_EXIT)
+                .addGeofence(this.buildLocationToTrack(scheduleWithLocationToTrack))
+                .build();
+    }
+
+    private Geofence buildLocationToTrack(Schedule scheduleWithLocationToTrack) {
+        GeoPoint locationCoords = scheduleWithLocationToTrack.getLocation();
+
+        return new Geofence.Builder()
+                .setRequestId(scheduleWithLocationToTrack.getUid())
+                .setCircularRegion(locationCoords.getLatitude(), locationCoords.getLongitude(),
+                        DEFAULT_TRACKING_RADIUS_METERS)
+                .setExpirationDuration(DEFAULT_TRACKING_TIME_MILLISECONDS)
+                .setNotificationResponsiveness(DEFAULT_MAX_NOTIFY_DELAY_MILLISECONDS)
+                .setTransitionTypes(Geofence.GEOFENCE_TRANSITION_ENTER | Geofence.GEOFENCE_TRANSITION_EXIT)
+                .build();
+    }
+
+    private PendingIntent getGeofencePendingIntent() {
+        if (geofencePendingIntent != null) {
+            return geofencePendingIntent;
+        }
+
+        Intent intent = new Intent(context, LocationBroadcastReceiver.class);
+        geofencePendingIntent = PendingIntent.getBroadcast(context, 0, intent,
```

Comment:
Refactor: Replace magic literal. 

The number 0 should have a meaning in this context, here it's unclear.

Reply:
- [ ] Completed?